src/error.rs

```diff
@@ -69,6 +71,43 @@ impl Error {
             _ => false,
         }
     }
+
+    /// Whether a read operation should be retried if this error occurs
+    pub(crate) fn is_read_retryable(&self) -> bool {
+        if self.is_network_error() {
+            return true;
+        }
+        let error_kind: &ErrorKind = &self.kind;
+        match &self.kind.code_and_message() {
+            Some((code, message)) => {
+                if RETRYABLE_READ_CODES.contains(&code) {
+                    return true;
+                }
+                if message.contains("not master") || message.contains("node is recovering") {
+                    return true;
+                }
+                false
+            }
+            None => false,
+        }
+    }
+
+    /// Whether an error originated in the driver
+    pub(crate) fn is_driver_error(&self) -> bool {
```

Comment:
The retryable reads spec indicates that if the retry fails with an error that originates within the driver rather than from the server, then the original error should be returned -- this list could use a double-check to make sure that nothing is erroneously included/excluded

Reply:
I think this would be easier to implement / maintain in the future by listing the server errors rather than driver ones, since the number of server errors is much smaller. I think the only server errors as of right now are `WriteError`, `CommandError`, `BulkWriteError`, and `AuthenticationError`.